driver/src/orderbook/streamed/orderbook.rs

```diff
@@ -65,16 +65,30 @@ impl Orderbook {
     }
 }
 
+fn filter_account_state(
+    account_states: impl Iterator<Item = ((UserId, TokenId), U256)>,
+    orders: &[Order],
+) -> AccountState {
+    let account_states = account_states
+        .filter(|((user, token), _)| {
+            orders.iter().any(|order| {
+                order.account_id == *user
+                    && (order.buy_token == *token || order.sell_token == *token)
```

Comment:
We are only expecting sellTokenBalances, so this check doesn't need to account for buy_tokens.